File: json/perl/read/json_read.pl
#! /usr/bin/perl
#
#	json_read.pl
#
#					Jul/02/2020
#
#
# ----------------------------------------------------------------
use	strict;
use	warnings;
use	utf8;
use	JSON;
use	Data::Dumper;
binmode STDOUT, ":encoding(UTF-8)";
binmode STDERR, ":encoding(UTF-8)";
#
use lib '/var/www/data_base/common/perl_common';
use text_manipulate;
use file_io;
#
# ----------------------------------------------------------------
print STDERR "*** 開始 ***\n";
my $file_in= $ARGV[0];
#
my $str_json = file_io::string_read_proc ($file_in);
#
my %dict_aa = %{decode_json ($str_json)};
#
text_manipulate::dict_display_proc (%dict_aa);
#
print STDERR "*** 終了 ***\n";
#
# ----------------------------------------------------------------

File: json/ruby/read/json_read.rb
#! /usr/bin/ruby
# -*- encoding: utf-8 -*-
#
#	json_read.rb
#
#					Nov/27/2017
#
# ---------------------------------------------------------------------
require 'json'
#
load '/var/www/data_base/common/ruby_common/file_io.rb'
load '/var/www/data_base/common/ruby_common/text_manipulate.rb'
# ---------------------------------------------------------------------
json_file = ARGV[0]
STDERR.puts	"*** 開始 ***"
#
json_str = read_string_proc(json_file)
#
dict_aa=JSON.parse(json_str)
#
dict_display_proc(dict_aa)
#
STDERR.puts	"*** 終了 ***"
#
# ---------------------------------------------------------------------
